### PR TITLE
Week calc module

### LINKED ISSUE #86: Turn calculate week_num into a module

### PR DESCRIPTION
**Pull request by AngelaGuardia**
### Code Highlights:
 * Abstract the week number calculation into a DateCalculator module to improve code DRYness
 * Update Era create and update mutations to use this module
 * ~~NOTE: waiting on events create and update PR #92 to be merged before finishing this~~
### Where should the reviewer start?
 * `/app/graphql/mutations/date_calculator.rb`
### Have Tests Been Added?
- [ ] No
- [ ] Yes, but all tests are not passing.
- [x] Yes, and all are passing.
### Issues:
* Closes #86
### Tracking Consistency:
- [x] added appropriate labels
- [x] My code follows the code style of this project and has removed all unnecessary annotations
- [ ] I have added comments on my pull request, particularly in hard-to-understand areas
- [x] looked at PR preview to check spelling, syntax, formatting, and completion
- [x] Rubocop Violations:

### CODE REVIEW

File: app/graphql/mutations/date_calculator.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module Mutations::DateCalculator
```
Comment (dhalverson) [inline app/graphql/mutations/date_calculator.rb:1]: Ahhhh, Mod1 is all coming back to me. This looks good

Comment (LHJE) [reply app/graphql/mutations/date_calculator.rb:1]: V good!  Thanks Angela!

File: app/graphql/mutations/eras/update_era.rb [hunk 2]
```diff
@@ -9,13 +11,15 @@ class Mutations::Eras::UpdateEra < ::Mutations::BaseMutation
 
   type Types::EraType
 
+  extend Mutations::DateCalculator
```
Comment (LHJE) [inline app/graphql/mutations/eras/update_era.rb:14]: Terrific!

File: spec/requests/graphql/mutations/date_calculator_spec.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      expect(week_num).to eq(1)
+
+      week_num = Mutations::DateCalculator.week_number(date_1y, user_bday)
+      expect(week_num).to eq(52)
+
+      week_num = Mutations::DateCalculator.week_number(date_90y, user_bday)
+      expect(week_num).to eq(4695)
+    end
+  end
+end
```
Comment (LHJE) [inline spec/requests/graphql/mutations/date_calculator_spec.rb:21]: Strong test, nicely done.

**Review by LHJE — APPROVED** [review]
Looks good to go!

File: app/graphql/mutations/events/create_event.rb [hunk 4]
```diff
@@ -1,19 +1,19 @@
...
-  module Events
-    class CreateEvent < Mutations::BaseMutation
-      argument :user_id, ID, required: true
-      argument :name, String, required: true
-      argument :date, String, required: true
-      argument :color, String, required: false
+require_relative '../date_calculator'
 
-      type Types::EventType
+class Mutations::Events::CreateEvent < Mutations::BaseMutation
```
Comment (leahriffell) [inline app/graphql/mutations/events/create_event.rb:3]: Thanks for getting this reformatting in too!

**Review by leahriffell — APPROVED** [review]
Niiice!

**Review by dhalverson — APPROVED** [review]
This looks good, nice job of drying this up.

**Pull request merged**